source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -69,11 +69,11 @@ void ShaderGraph::addOutputSockets(const InterfaceElement& elem)
     }
 }
 
-void ShaderGraph::createConnectedNodes(ElementPtr downstreamElement,
-                                        ElementPtr upstreamElement,
-                                        ElementPtr connectingElement,
-                                        GenContext& context,
-                                        ShaderNode* rootNode)
+void ShaderGraph::createConnectedNodes(const ElementPtr& downstreamElement,
+                                       const ElementPtr& upstreamElement,
+                                       ElementPtr connectingElement,
```

Comment:
Nitpick: The signature looks a bit suspicious when the connectingElement is not const while the others are.. I understand why by reading the full function code, but it makes me think, should this code really be extracted to its own function? It is only called from one place so it's extra overhead, and to my mind it was more readable before when it was just a code block instead of a separate function.